apps/src/code-studio/components/header/SignInCalloutLogic.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import cookies from 'js-cookie';
+import SignInCallout from './SignInCallout';
+
+const HideSignInCallout = 'hide_signin_callout';
+
+// The use of both session storage and cookies is to check for 1 day
+// and 1 session, and display the callout again only once BOTH have passed.
+export default class SignInCalloutLogic extends React.Component {
```

Comment:
It seems weird to me to have `Logic` as part of a component name, I'm trying to brainstorm better names... `SignInCalloutWrapper`? or just `SignInCallout` and rename the other component? Open to other suggestions

Reply:
'Wrapper' works for me.